runtime/src/main/cpp/Memory.cpp

```diff
@@ -3439,6 +3439,13 @@ void MutationCheck(ObjHeader* obj) {
     ThrowInvalidMutabilityException(obj);
 }
 
+void CheckLifetimesConstraint(ObjHeader* obj, ObjHeader* pointee) {
+  if (!obj->local() && pointee != nullptr && pointee->local()) {
+    konan::consolePrintf("Attempt to store a stack object %p into a heap object %p\n", pointee, obj);
```

Comment:
It also makes sense to make this message more clear to the users. E.g. to add
`This is a compiler bug, please report it to https://kotl.in/issue`.